### ISSUE TITLE
Replace in file

### ISSUE THREAD

**Issue by frederikhors** [vladshcherbin/rollup-plugin-copy#21]
It would be amazing to replace some vars in files using something like https://github.com/adamreisnz/replace-in-file.

Example:

- I have the `index.html` in `public` folder
- I start with `npm run dev`
- This plugin copy `index.html` from `public` to `dist`
- After the copy or in memory change the `%BundleNameVarFromRollup%` value in `<script src="/%BundleNameVarFromRollup%.js"/>`

Is it possible?

**Comment by vladshcherbin** [comment]
I'd searched for a plugin to change contents of a file so the workflow is this:

- npm run dev
- files copied to `dist`
- plugin replaces contents in this `dist` `index.html` file

or another plugin to process `index.html` file separately

- npm run dev
- files copied to `dist`, but not `index.html`
- `index.html` is processed with another plugin and saved in `dist` folder

May help:

https://www.npmjs.com/package/rollup-plugin-string-replacer
https://www.npmjs.com/package/rollup-plugin-replace-html-vars

What do think about this options?

Alternatively, in the next version a target object will have `src`, `dist` and `rename` properties. I can think about adding replace function with similar functionality if nothing else works 🤔

**Comment by frederikhors** [comment]
@vladshcherbin I'm happy to see you answer right away.

My needs are fairly basic.

I could do all the operations with external cli, eliminating the use of rollup plugins.

But I saw that yours is great and I would like to expand it with these features which, I repeat, I consider basic for the development of a web application.

Now I'm using https://github.com/haifeng2013/rollup-plugin-bundle-html but as you can see in the issues there are a few problems.

In general it would be useful that in the development phase (without hash in file name) there is an index that is written (even only the first time) and that contains the .js and .css inject.

In the production phase obviously the bundles will have the hash in the name and therefore inject is with the hash.

I think your plugin needs to extend the functionality by adding this.

Thank you for your work.

**Comment by vladshcherbin** [comment]
@frederikhors how about a `transform` option to give you contents or filename so you can use functions/packages to change it? Something like this:

```js
copy({
  targets: {
    src: 'src/index.html',
    dest: 'dist',
    transform: contents => contents.replace('%BundleNameVarFromRollup%', 'app.js')
  }
})
```

**Comment by frederikhors** [comment]
Yes, something like this.

I need to use src without hash in file name in development and hashes in production bundles.

**Comment by frederikhors** [comment]
@vladshcherbin is it possible to get the hash of the file from rollup? Maybe choosing a final hook?

**Comment by vladshcherbin** [comment]
@frederikhors using a separate plugin I believe, currently no tools for this in `rollup-plugin-copy`

**Comment by frederikhors** [comment]
@vladshcherbin Closing this issue would make `rollup-plugin-copy` essential to any rollup project. The non-plus-ultra.

**Comment by JdesEva** [comment]
Unable to copy file to destination folder
![image](https://user-images.githubusercontent.com/37232561/69791690-01ef9580-1200-11ea-9188-0b6e7e05641e.png)

**Comment by JdesEva** [comment]
> Unable to copy file to destination folder
> ![image](https://user-images.githubusercontent.com/37232561/69791690-01ef9580-1200-11ea-9188-0b6e7e05641e.png)

OK，I know.This must use an absolute path

**Comment by CPatchane** [comment]
Is https://github.com/vladshcherbin/rollup-plugin-copy/pull/31 fixing the issue here?

**Comment by vladshcherbin** [comment]
@CPatchane I believe so, will check later this week, thank you

**Comment by vladshcherbin** [comment]
Transform feature was released in 3.3.0 with the help of @CPatchane 🎉
